include/clink/kernels/opdefs/clink_kernels.td

```diff
@@ -87,39 +91,40 @@ def SquareF64Op: Clink_Op<"square.f64"> {
   let verifier = ?;
 }
 
-def OneHotEncoderLoadOp: Clink_Op<"onehotencoder_load"> {
-  let summary = "onehotencoder_load operation";
+class LoadOp<string suffix> : Clink_Op<"load." # suffix> {
+  let summary = "clink.load operation";
   let description = [{
-     An operation that loads a OneHotEncoderModel from a given path. The path should be a directory
-     containing params and model data saved through
-     org.clink.feature.onehotencoder.ClinkOneHotEnoderModel::save(...).
+     An operation that creates a Model's subclass instance based on the 
+     data read from the given path. The path should be a directory
+     containing params and model data saved by Clink's corresponding Java 
+     operator's void save(String path) method.
 
      Example:
-       %1 = clink.onehotencoder_load %0
+       %1 = clink.load.onehotencoder %0
   }];
   let arguments = (ins TFRT_StringType:$path);
   let results = (outs Clink_ModelType:$model);
   let assemblyFormat = "operands attr-dict";
   let verifier = ?;
 }
 
-// TODO: Change this to generic kernels for load and transform operations
-def OneHotEncoderTransformOp: Clink_Op<"onehotencoder_transform"> {
-  let summary = "onehotencoder_transform operation";
+def OneHotEncoderLoadOp : LoadOp<"onehotencoder">;
+
+def TransformOp: Clink_Op<"transform"> {
+  let summary = "transform operation";
   let description = [{
-     An operation that transforms data based on a OneHotEncoderModel. It takes a value
-     and its index among all columns in a database table that needs to be one-hot
-     encoded, and returns the resulting encoded sparse vector.
+     An operation that transforms data based on a Model. It applies the Model 
+     on the given ArrayRef of input AsyncValues and returns the result Async 
+     values as a SmallVector.
 
      Example:
-       %3 = clink.onehotencoder_transform %0, %1, %2
+       %2 = clink.transform %0, %1
```

Comment:
Instead of taking an array ref as input and return a small vector as output, could we let this kernel take `Variadic<AnyType>` (i.e. `Object...` in Java) as input and return `Variadic<AnyType>` as output. Please see `tfrt_data.enumerate.iterator` for how to use `Variadic<AnyType>` to define kernels.

This approach is more consistent with TFRT's style (e.g. `tfrt_data.iterator_get_next` and `tfrt_data.enumerate.iterator`) where values are directly exposed in TFRT instead of being wrapped in a vector. This approach gives MLIR more information of kernels's input/output, which could be useful for graph optimization.

And we probably don't need `ArrayRefType` and `SmallVectorType` if we follow this approach.

Reply:
I agree with the idea to avoid wrapping input and output values in a vector. 

So far as I can see, we need to use `RemainingArguments` and `RemainingResults` as the new input and output values. Does it mean we should also modify the `Model::transform()` API? For example, `llvm::Error transform(RemainingArguments, RemainingResults)`?

Reply:
Thanks for the explanation.

Reply:
As discussed offline, `Model::transform()` does not need to take `RemainingArguments` as input. The kernel `clink.transform` could translate `RemainingArguments` into the type accepted by `Model::transform()`.